### PR TITLE
feat(fs): support reading file ranges

### PR DESCRIPTION
**Pull request by tusharmath**
This PR introduces significant enhancements to file handling capabilities with focus on large file support and accurate binary detection.

### Range-based File Reading
- Add specialized support for reading specific byte ranges from large text files
- Implement UTF-8 boundary detection for safe text handling 
- Optimize memory usage by reading only the requested portions of files
- Update FSReadInput structure with start_byte and end_byte fields

### Improved Binary File Detection
- Integrate the infer crate for reliable file type identification using signatures
- Implement multi-step detection: signature-based, statistical analysis, and UTF-8 validation
- Enhance accuracy with better handling of edge cases including empty files
- Add comprehensive test coverage for various file types

### UTF-8 Safety Improvements
- Change file read services to return UTF-8 strings instead of raw bytes
- Implement proper error handling for invalid UTF-8 sequences
- Fix boundary handling to ensure character integrity

## Technical Implementation
- Added new modules in forge_fs for specialized functionality
- Updated service interfaces for backward compatibility
- Included proper dependency management with cfb and infer packages
- Added detailed documentation and tests

> Note: This PR was originally titled "test auto tracking" to evaluate commit tracking mechanisms.

### CODE REVIEW

File: crates/forge_services/src/attachment.rs [hunk 1]
```diff
@@ -43,20 +43,23 @@ impl<F: Infrastructure> ForgeChatRequest<F> {
                 .cwd
                 .join(path)
         }
-        let read = self.infra.file_read_service().read(path.as_path()).await?;
+        let file_content = self.infra.file_read_service().read(path.as_path()).await?;
```
Comment (tusharmath) [inline crates/forge_services/src/attachment.rs]: Need to use `read_range` over here as well. Keeping the format like in fs_read.

File: crates/forge_services/src/attachment.rs [hunk 2]
```diff
@@ -141,13 +144,55 @@ pub mod tests {
...
+                    .take((end - start) as usize)
+                    .collect()
+            } else {
+                String::new()
+            };
+
+            Ok((
+                range_content,
+                forge_fs::FileInfo::new(start, end, total_size),
+            ))
```
Comment (tusharmath) [inline crates/forge_services/src/attachment.rs:187]: Can keep it as unimplemented or proxy it to `ForgeFS`.

File: crates/forge_fs/src/read_range.rs [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    /// Returns a tuple containing:
+    /// - The file content as a UTF-8 string.
+    /// - FileInfo containing metadata about the read operation including
+    ///   character positions.
+    pub async fn read_range_utf8<T: AsRef<Path>>(
+        path: T,
+        start_char: Option<u64>,
+        end_char: Option<u64>,
+    ) -> Result<(String, FileInfo)> {
+        let path_ref = path.as_ref();
```
Comment (tusharmath) [inline crates/forge_fs/src/read_range.rs:21]: We should create a file handle in this function, and pass than handle to all the utility functions so that we don't keep opening the file again and again.
